src/buildkite/agent.go

```diff
@@ -0,0 +1,72 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	osexec "golang.org/x/sys/execabs"
+)
+
+type Agent struct {
+}
+
+func (a Agent) Annotate(ctx context.Context, message string, style string, annotationContext string) error {
+	return execCmdWithStdin(ctx, "buildkite-agent", message, "annotate", "--style", style, "--context", annotationContext)
+}
+
+func (a Agent) ArtifactUpload(ctx context.Context, path string) error {
+	return execAgentSyscall(ctx, "buildkite-agent", "artifact", "upload", path)
+}
+
+func execAgentSyscall(ctx context.Context, executableName string, args ...string) error {
+	Logf("Executing: %s %s\n", executableName, strings.Join(args, " "))
+
+	executable, err := osexec.LookPath(executableName)
+	if err != nil {
+		return err
+	}
+
+	execArgs := append([]string{executableName}, args...)
+	return syscall.Exec(executable, execArgs, os.Environ())
```

Comment:
I just noticed the glob pattern being used here: https://github.com/cultureamp/ecr-scan-results-buildkite-plugin/pull/10/files#diff-9e185f29fa355d7dd8fdd9c9ff1d0723b85206aa7d37c4eec93997005dc291ebR143

I'm guessing it's so that works.

Reply:
No, it's mostly about it being more straightforward etc. The globbing is passed directly to the agent for interpretation as far as I can tell.